runtime/lua/vim/lsp/util.lua

```diff
@@ -1638,29 +1672,33 @@ function M.try_trim_markdown_code_blocks(lines)
   return 'markdown'
 end
 
-local str_utfindex = vim.str_utfindex
 ---@private
-local function make_position_param()
-  local row, col = unpack(api.nvim_win_get_cursor(0))
+---@param window (optional, number): window handle or 0 for current, defaults to current
+---@param encoding string utf-8|utf-16|utf-32|nil defaults to utf-16
+local function make_position_param(window, encoding)
+  window = window or 0
```

Comment:
Shouldn't this be "buffer" (semantically speaking) since the encodings are per buffer not per window?

Reply:
The way I saw it was this function gets the cursor position from a window, so passing in a buffer could result in a mismatch.